indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskStorageQueryAdapter.java

```diff
@@ -104,7 +104,14 @@ public List<TaskStatusPlus> getTaskStatusPlusList(
 
   public Optional<Task> getTask(final String taskid)
   {
-    return storage.getTask(taskid);
+    // Try to fetch active task from memory
+    final Task activeTask = taskLockbox.getActiveTask(taskid);
```

Comment:
I thought this should always be called under the giant lock no ?

Reply:
Thanks, I've ensured that the call happens within a lock in the new TaskQueue method